### ISSUE TITLE
Unable to mount USB device after supervisor update

### ISSUE THREAD

**Issue by doosterkamp** [home-assistant/supervisor#2590]
Yesterday my HA updated the Supervisor to *'supervisor-2021.02.11'*.  
After a reboot today I **can't** mount my USB storage anymore:  

  `mount: mounting /dev/sda1 on /share/usb failed: Permission denied`  

Before yesterday, it worked as expected.  
  
I have an EXT4 formatted USB thumb drive connected to my RPi 2.  
I use a slightly modified version of the MariaDB addon, to host the database on the thumb drive and prevent issues with my SD-Card. Until yesterday, this worked like a charm (even though I had to disable the 'Protection mode')!
  
I don't think using a thumb drive as an extra storage is a security threat anyway, so blocking it in protection mode is overprotecting the system imho.

Please enable access to my usb drive (/dev/sda*) again.  
I can live with disabling 'Protection Mode' or is there an other way of mounting it, that I'm not thinking of or am aware of?

**Comment by doosterkamp** [comment]
Also: downgrading Supervisor isn't working either  
  
` ha > supervisor update --version 2021.02.10 `  
Still running 2021.02.11 (also after restarting and rebooting)  
  
` ha > supervisor update --version 2021.02.9 `  
same result ...

**Comment by ms32035** [comment]
Looks like a wider issue, as after the update my addon which has `full_access` lost access to `/proc/device-tree/system` with protected mode disabled

**Comment by curlydingo** [comment]
I think it's due to [this](https://github.com/home-assistant/supervisor/commit/b8a976b3440c57bc6cd3f8617afcc632f57906ff#diff-9d9b42663129dffdf9ab9475772d2161c835d56d51ba861d73d93e9340980e74R232) commit, which changed the /dev mount from rw to ro. Seems to be breaking a few things, including [zigbee2mqtt](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/24). I think this would affect any addon that uses devices directly.

Looking through the commit and it's pull request I can't see any mention of why it changed to read-only though.

**Comment by adamoutler** [comment]
This is causing a problem for enablement of I2C for new users of Argon One.  Mounting  /dev/sda1 is required to enable I2C and allow fan control.  Without I2C there is no way to regulate the temperature on these devices.   The only option left is to enable I2C via [Home Assistant Operating System Terminal](https://community.home-assistant.io/t/add-on-hassos-i2c-configurator/264167/70?u=adamoutler).  While I'm very glad I wrote those instructions, because they are now the only ones that work on devices with built-in hard drives, it's not user friendly and the users of my Enable I2C addon are not the type who would type a large block of text.   See [angry users in forum](https://community.home-assistant.io/t/add-on-hassos-i2c-configurator/264167/70?u=adamoutler) 

The ArgonOne is the most popular Hard Drive case for Raspberry Pi right now. Both the Argon One and DeskPi Pro use a proprietary U-shaped USB-A to USB-A connector for the hard drive.  There is no way to plug these devices into a computer.  The only solution would be to create a dangerous USB-A to USB-A cable which, if wired wrong or plugged into the wrong device, would damage the computer. 

Similarly, this affects Enabling Host-Mode Serial on devices with built-in hard drives. Which means DeskPi Pro cannot activate it's built-in Active Cooling system on Home Assistant Operating System, nor utilize it's front-facing USB ports. 

Additionally this affects the ability to enable [Home Assistant OS Port 22222 SSH](https://developers.home-assistant.io/docs/operating-system/debugging/).  I think this is a pretty serious blocking issue.

**Comment by adamoutler** [comment]
Looks like this is going to be a wont-fix status  https://github.com/home-assistant/supervisor/pull/2596

**Comment by pvizeli** [comment]
back to the topic. Add-on created with supervisor-2021.02.11 should give access to disk again. Before there was a bug. Can you post the add-ons `/sys/fs/cgroup/devices/devices.list`? Of course, the device needs to be inside the add-on config or with the new device options

**Comment by doosterkamp** [comment]
I don't know where to find the add-ons `/sys/fs/cgroup/devices/devices.list`, but the system one I see -using the Terminal- looks like this:
```
b *:* m
c *:* m
c 1:3 rwm
c 1:5 rwm
c 1:7 rwm
c 1:8 rwm
c 1:9 rwm
c 5:0 rwm
c 5:1 rwm
c 5:2 rwm
c 10:200 rwm
c 136:* rwm
```


My add-on's config.json:
```
{
  "name": "MariaDB on USB",
  "version": "2.2.1.0",
  "slug": "mariadb",
  "description": "An SQL database server, hosted on an USB thumb drive",
  "url": "https://github.com/doosterkamp/MariaDB-on-USB/tree/master/mariadb",
  "arch": ["armhf", "armv7", "aarch64", "amd64", "i386"],
  "startup": "system",
  "init": false,
  "timeout": 20,
  "services": ["mysql:provide"],
  "ports": {
    "3306/tcp": 3306
  },
  "options": {
    "databases": ["homeassistant"],
    "logins": [{ "username": "homeassistant", "password": null }],
    "rights": [
      {
        "username": "homeassistant",
        "database": "homeassistant"
      }
    ]
  },
  "schema": {
    "databases": ["str"],
    "logins": [{ "username": "str", "password": "str" }],
    "rights": [{ "username": "str", "database": "str" }]
  },
  "devices": ["/dev/sda1"]
}
```

Within the add-on `mariadb\rootfs\etc\services.d\mariadb\run`:

```
#!/usr/bin/with-contenv bashio
# ==============================================================================
# Start MariaDB service if enabled
# ==============================================================================
USB_BASE=/share/usb

MARIADB_DATA=${USB_BASE}/databases
MARIADB_SECRET=${USB_BASE}/secret
NEW_INSTALL=false

if [[ ! -e /share/usb ]]; then
    bashio::log.info "Create mount point for USB disk..."
    mkdir -p /share/usb
    chmod -R 0777 /share/usb
fi

if [[ -e /dev/sda1 ]]; then
    bashio::log.info "Mounting USB disk..."
    mount /dev/sda1 /share/usb
fi

...
```

And finally, I asume you anly need to see the sda part:

```
  - name: sda
    sysfs: >-
      /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.3/1-1.3:1.0/host0/target0:0:0/0:0:0:0/block/sda
    dev_path: /dev/sda
    subsystem: block
    by_id: '/dev/disk/by-id/usb-Samsung_Flash_Drive_0373520010000370-0:0'
    attributes:
      DEVLINKS: >-
        /dev/disk/by-id/usb-Samsung_Flash_Drive_0373520010000370-0:0
        /dev/disk/by-path/platform-3f980000.usb-usb-0:1.3:1.0-scsi-0:0:0:0
      DEVNAME: /dev/sda
      DEVPATH: >-
        /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.3/1-1.3:1.0/host0/target0:0:0/0:0:0:0/block/sda
      DEVTYPE: disk
      ID_BUS: usb
      ID_INSTANCE: '0:0'
      ID_MODEL: Flash_Drive
      ID_MODEL_ENC: Flash\x20Drive\x20\x20\x20\x20\x20
      ID_MODEL_ID: '1000'
      ID_PART_TABLE_TYPE: gpt
      ID_PART_TABLE_UUID: 0d6e14ae-5858-9090-8081-828310111213
      ID_PATH: 'platform-3f980000.usb-usb-0:1.3:1.0-scsi-0:0:0:0'
      ID_PATH_TAG: platform-3f980000_usb-usb-0_1_3_1_0-scsi-0_0_0_0
      ID_REVISION: '1100'
      ID_SERIAL: 'Samsung_Flash_Drive_0373520010000370-0:0'
      ID_SERIAL_SHORT: '0373520010000370'
      ID_TYPE: disk
      ID_USB_DRIVER: usb-storage
      ID_USB_INTERFACES: ':080650:'
      ID_USB_INTERFACE_NUM: '00'
      ID_VENDOR: Samsung
      ID_VENDOR_ENC: Samsung\x20
      ID_VENDOR_ID: 090c
      MAJOR: '8'
      MINOR: '0'
      SUBSYSTEM: block
      TAGS: ':systemd:'
      USEC_INITIALIZED: '7799528'
  - name: sda1
    sysfs: >-
      /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.3/1-1.3:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda1
    dev_path: /dev/sda1
    subsystem: block
    by_id: '/dev/disk/by-id/usb-Samsung_Flash_Drive_0373520010000370-0:0-part1'
    attributes:
      DEVLINKS: >-
        /dev/disk/by-id/usb-Samsung_Flash_Drive_0373520010000370-0:0-part1
        /dev/disk/by-label/usb /dev/disk/by-partlabel/Basic\x20data\x20partition
        /dev/disk/by-partuuid/dfe6ac2e-751b-11eb-987d-90ccdfbe619d
        /dev/disk/by-path/platform-3f980000.usb-usb-0:1.3:1.0-scsi-0:0:0:0-part1
        /dev/disk/by-uuid/0F58-0B94
      DEVNAME: /dev/sda1
      DEVPATH: >-
        /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.3/1-1.3:1.0/host0/target0:0:0/0:0:0:0/block/sda/sda1
      DEVTYPE: partition
      ID_BUS: usb
      ID_FS_BLOCK_SIZE: '512'
      ID_FS_LABEL: usb
      ID_FS_LABEL_ENC: usb
      ID_FS_LABEL_FATBOOT: usb
      ID_FS_LABEL_FATBOOT_ENC: usb
      ID_FS_TYPE: vfat
      ID_FS_USAGE: filesystem
      ID_FS_UUID: 0F58-0B94
      ID_FS_UUID_ENC: 0F58-0B94
      ID_FS_VERSION: FAT32
      ID_INSTANCE: '0:0'
      ID_MODEL: Flash_Drive
      ID_MODEL_ENC: Flash\x20Drive\x20\x20\x20\x20\x20
      ID_MODEL_ID: '1000'
      ID_PART_ENTRY_DISK: '8:0'
      ID_PART_ENTRY_NAME: Basic\x20data\x20partition
      ID_PART_ENTRY_NUMBER: '1'
      ID_PART_ENTRY_OFFSET: '2048'
      ID_PART_ENTRY_SCHEME: gpt
      ID_PART_ENTRY_SIZE: '250611951'
      ID_PART_ENTRY_TYPE: ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
      ID_PART_ENTRY_UUID: dfe6ac2e-751b-11eb-987d-90ccdfbe619d
      ID_PART_TABLE_TYPE: gpt
      ID_PART_TABLE_UUID: 0d6e14ae-5858-9090-8081-828310111213
      ID_PATH: 'platform-3f980000.usb-usb-0:1.3:1.0-scsi-0:0:0:0'
      ID_PATH_TAG: platform-3f980000_usb-usb-0_1_3_1_0-scsi-0_0_0_0
      ID_REVISION: '1100'
      ID_SERIAL: 'Samsung_Flash_Drive_0373520010000370-0:0'
      ID_SERIAL_SHORT: '0373520010000370'
      ID_TYPE: disk
      ID_USB_DRIVER: usb-storage
      ID_USB_INTERFACES: ':080650:'
      ID_USB_INTERFACE_NUM: '00'
      ID_VENDOR: Samsung
      ID_VENDOR_ENC: Samsung\x20
      ID_VENDOR_ID: 090c
      MAJOR: '8'
      MINOR: '1'
      PARTN: '1'
      PARTNAME: Basic data partition
      SUBSYSTEM: block
      TAGS: ':systemd:'
      USEC_INITIALIZED: '7914415'
      dir_name: usb
      mount_point: /mnt/data/supervisor/share/usb
```

**Comment by pvizeli** [comment]
I need at minimum the output of the add-ons `/sys/fs/cgroup/devices/devices.list`, if they have `b 8:1 rwm`, you have full access to this device node

**Comment by doosterkamp** [comment]
Darn ...
I found a much better solution for what I need, nicely hidden in your operating-system documentation: https://github.com/home-assistant/operating-system/blob/master/Documentation/partition.md

I'm moving the complete data partition to the thumb drive, so my sd is used only RO after initial setup.

This should be an option during setup!
I've been searching for this option, since I started using HA and found out about this just 5 minutes ago, after many many many different Google queries. 

**Not many people know about this option!**

I've read tons of complex tutorials on how to mount it using udev rules or add-on configuration edits, while this is THE solution for all the Raspberry Pi users, that have experienced it's major weakness: killing SD cards by 'many' write actions (like databases do)

I'll close this issue and will post the link to the mentioned move tutorial in every discussion/tutorial I've come around.
Thanks for your effort and attention!
